doc/source/tune/_tutorials/tune-serve-integration-mnist.py

```diff
@@ -0,0 +1,645 @@
+    best_trial_config = best_trial.config
+    best_checkpoint = best_trial.checkpoint.value
+
+    return best_accuracy, best_trial_config, best_checkpoint, N
+
+
+#######################################################################
+# Serving tuned models with Ray Serve
+# -----------------------------------
+# Let's now turn to the model serving part with Ray Serve. Serve
+# distinguishes between _backends_ and _endpoints_. Broadly speaking, a
+# backend handles incoming requests and replies with a result. For
+# instance, our MNIST backend takes an image as input and outputs the
+# digit it recognized from it. An endpoint on the other hand forwards
+# incoming HTTP requests to one or more different backends, according
+# to a routing policy.
+#
+# First, we will define our backend. This backend loads our PyTorch
+# MNIST model from a checkpoint, takes an image as an input and
+# outputs our digit prediction according to our trained model:
+class MNISTBackend:
+    def __init__(self, checkpoint_dir, config, metrics):
+        self.checkpoint_dir = checkpoint_dir
+        self.config = config
+        self.metrics = metrics
+
+        # use_cuda = config.get("use_gpu") and torch.cuda.is_available()
+        # self.device = torch.device("cuda" if use_cuda else "cpu")
+        self.device = "cpu"
+        model = ConvNet(layer_size=self.config["layer_size"]).to(self.device)
+
+        model_state, optimizer_state = torch.load(
+            os.path.join(self.checkpoint_dir, "checkpoint"))
+        model.load_state_dict(model_state)
+
+        self.model = model
+
+    def __call__(self, flask_request):
+        images = torch.tensor(flask_request.json["images"])
+        images = images.to(self.device)
+        outputs = self.model(images)
+        predicted = torch.max(outputs.data, 1)[1]
+        return {"result": predicted.numpy().tolist()}
+
+
+#######################################################################
+# We would like to have a fixed location where we store the currently
+# active model. We call this directory ``model_dir``. Everytime we
+# would like to update our model, we copy the checkpoint of the new
+# model to this directory. We then create a new backend pointing to
+# that directory, route all the traffic on our model endpoint to this
+# backend, and then delete the old backends to free up some memory.
+def serve_new_model(model_dir, checkpoint, config, metrics, day):
+    print("Serving checkpoint: {}".format(checkpoint))
+
+    if not os.path.exists(model_dir):
+        try:
+            os.mkdir(model_dir, 0o755)
+        except OSError:
+            return False
+
+    checkpoint_path = os.path.join(model_dir, "checkpoint")
+    meta_path = os.path.join(model_dir, "meta.json")
+
+    if os.path.exists(checkpoint_path):
+        shutil.rmtree(checkpoint_path)
+
+    shutil.copytree(checkpoint, checkpoint_path)
+
+    with open(meta_path, "wt") as fp:
+        json.dump(dict(config=config, metrics=metrics), fp)
```

Comment:
can you move these to a helper function if possible? It's a bit distracting for audience trying figure out which part is doing the serving and which part is just writing checkpoints.